checker/src/expression.rs

```diff
@@ -669,6 +680,7 @@ impl Expression {
                 operand.expression.contains_local_variable()
             }
             Expression::Reference(path) => path.contains_local_variable(),
+            Expression::RefinedVariable { .. } => false,
```

Comment:
Why is this cases treated differently from Variable?

Reply:
Only paths rooted in parameters can get wrapped up with RefinedVariable.

Reply:
https://github.com/facebookexperimental/MIRAI/blob/56719f2b662bdc8b0fcaa62f3b7244d0d018c9bd/checker/src/abstract_value.rs#L2881-L2884
I was assuming that but I got less confident when I read the code above.

BTW It would be nicer if this invariant for RefinedVariable is documented.